opendut-edgar/src/service/can_manager.rs

```diff
@@ -0,0 +1,213 @@
+use std::net::IpAddr;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::{Arc, Mutex};
+use regex::Regex;
+
+use tokio::process::Command;
+
+use opendut_types::util::net::NetworkInterfaceName;
+
+use crate::service::cannelloni_manager::CannelloniManager;
+use crate::service::network_interface::manager::NetworkInterfaceManagerRef;
+
+pub type CanManagerRef = Arc<CanManager>;
+
+pub struct CanManager{
+    cannelloni_termination_token: Mutex<Arc<AtomicBool>>,
```

Comment:
From my understanding, the Mutex here is needed because the `AtomicBool` is wrapped by an `Arc`. The `Arc` in turn seems to be needed because the `CannelloniManager`s will still need to access the `AtomicBool` after the `CanManager` already replaced its `cannelloni_termination_token` with a new one.

Reply:
If we reuse the same `AtomicBool` and just assign it `false`, there is a good chance that the `CannelloniManager`s do not read the `AtomicBool`in the short time where it's `true`, causing them to not notice that they should terminate. That's the reason why I create a new `AtomicBool`to use for termination requests of the next group of `CannelloniManager`s and leave the old `AtomicBool` to be read by all old `CannelloniManager`s so that they can terminate as intended.

Reply:
As agreed during today's meeting, I now documented this mechanism in the code.